tools/update_lockfiles.py

```diff
@@ -0,0 +1,55 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+A command line utility for generating conda-lock files for the environments
+that nox uses for testing each different supported version of python.
+Typical usage:
+
+    python tools/update_lockfiles.py -o requirements/ci/nox.lock requirements/ci/py*.yml
+
+
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+
+
+try:
+    import conda_lock
+except:
+    print("conda-lock must be installed.")
+    exit(1)
+
+parser = argparse.ArgumentParser(
+    "Iris Lockfile Generator",
+)
+
+parser.add_argument('files', nargs='+', 
+    help="List of environment.yml files to lock")
+parser.add_argument('--output-dir', '-o', default='.', 
+    help="Directory to save output lock files")
+
+args = parser.parse_args()
+
+for infile in args.files:
+    print(f"generating lockfile for {infile}", file=sys.stderr)
+    fname = os.path.basename(infile)
```

Comment:
Switch to `Pathlib` if possible.

Reply:
Interesting! What are the benefits of using `pathlib` over `os.path`?  I'll be honest, I'm not super keen on the overloading of the `__div__` operator

Reply:
@pp-mo presented it to me as the modern way to go, and TBH I have found it nice and intuitive through lots of small conveniences. [No doubt there are myriad reasons](https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/), but two that spring to mind from personal experience:

* The availability of `Path.glob()`
* Being able to write `mypath / subdir / subsubdir`